pkg/api/defaults_test.go

```diff
@@ -0,0 +1,101 @@
+package api
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func validOpenShiftClusterDocument() *OpenShiftClusterDocument {
+	doc := OpenShiftClusterDocument{
+		OpenShiftCluster: &OpenShiftCluster{
+			Properties: OpenShiftClusterProperties{
+				NetworkProfile: NetworkProfile{
+					SDNProvider: SDNProviderOpenShiftSDN,
+				},
+				MasterProfile: MasterProfile{
+					EncryptionAtHost: EncryptionAtHostDisabled,
+				},
+				WorkerProfiles: []WorkerProfile{
+					{
+						EncryptionAtHost: EncryptionAtHostDisabled,
+					},
+				},
+			},
+		},
+	}
+
+	return &doc
+}
+func TestSetDefaults(t *testing.T) {
+	for _, tt := range []struct {
+		name  string
+		want  func() *OpenShiftClusterDocument
+		input func(doc *OpenShiftClusterDocument)
+	}{
+		{
+			name: "no defaults needed",
+			want: func() *OpenShiftClusterDocument {
+				doc := validOpenShiftClusterDocument()
+				return doc
+			},
+			input: func(base *OpenShiftClusterDocument) {
+			},
```

Comment:
```suggestion
			want: func() *OpenShiftClusterDocument {
				return validOpenShiftClusterDocument()
			},
```

Note: removes `input`, so need to do `if tt.input != nil` check below before calling it.